ci/glci/az.py

```diff
@@ -0,0 +1,464 @@
+
+        response = requests.get(
+            self._api_url(publisher_id, "offers", offer_id, "submissions"),
+            headers=self.req_headers,
+            params=self.req_params,
+        )
+        if not response.ok:
+            raise RuntimeError("Could not fetch Azure Marketplace transport operations for gardenlinux offer.")
+        operations = response.json()
+        response.close()
+        for operation in operations:
+            if AzmpTransportDest(operation["slot"]) == transport_dest and AzmpOperationState(operation["submissionState"]) == AzmpOperationState.RUNNING:
+                return operation["id"]
+        raise RuntimeError(f"Did not find an ongoing transport operation to ship gardenliunx offer on the Azure Marketplace.")
+
+    def fetch_operation_state(self, publisher_id: str, offer_id: str, operation_id: str):
+        """Fetches the state of a given Azure Marketplace transport operation."""
+
+        response = requests.get(
+            self._api_url(publisher_id, "offers", offer_id, "operations", operation_id),
+            headers=self.req_headers,
+            params=self.req_params,
+        )
+        if not response.ok:
+            raise RuntimeError(f"Can't fetch state for transport operation {operation_id}. statuscode={response.status_code}")
+        operation = response.json()
+        response.close()
+        return AzmpOperationState(operation['status'])
+
+    def go_live(self, publisher_id: str, offer_id: str):
+        """Trigger a go live operation to transport an Azure Marketplace offer to production."""
+
+        response = requests.post(
+            self._api_url(publisher_id, "offers", offer_id, "golive"),
+            headers=self.req_headers,
+            params=self.req_params,
+        )
+        if not response.ok:
+            raise RuntimeError(
+                    "Go live of updated gardenlinux Azure Marketplace offer failed."
+                    f"statuscode={response.status_code}"
+                )
+
+
+def add_image_version_to_plan(
+    spec: dict,
+    plan_id: str,
+    image_version: str,
+    image_url: str
+):
+    """Add a new image version to a given plan and return a modified offer spec."""
+
+    plan_spec = {}
+    for plan in spec["definition"]["plans"]:
+        if plan["planId"] == plan_id:
+            plan_spec = plan
+
+    if not plan_spec:
+        raise RuntimeError(f"Plan {plan_id} not found in offer {spec['id']}.")
+
+    plan_spec["microsoft-azure-virtualmachines.vmImages"][image_version] = {
+        "osVhdUrl": image_url,
+        "lunVhdDetails": []
+    }
+    return spec
+
+
+def remove_image_version_from_plan(spec: dict, plan_id: str, image_version: str, image_url: str):
+    """remove an image version from a given plan and return a modified offer spec."""
+
+    plan_spec = {}
+    for plan in spec["definition"]["plans"]:
+        if plan["planId"] == plan_id:
```

Comment:
since this is now the second implementation of plan-lookup, I would actually suggest to factor this out; maybe it could be included into the function or method that returns the initial definition-dict

Reply:
done